pandas/formats/style.py

```diff
@@ -851,39 +851,178 @@ def set_properties(self, subset=None, **kwargs):
+                     ', transparent {zero:.1f}%, {c} {zero:.1f}%, '
+                     '{c} {w:.1f}%, transparent {w:.1f}%)')
+
+        return [attrs_pos.format(c=color[1], zero=zero, w=x) if x > zero
+                else attrs_neg.format(c=color[0], zero=zero, w=x)
+                for x in normed]
+
+    def bar(self, subset=None, align='left', axis=0,
+            color='#d65f5f', width=100):
         """
         Color the background ``color`` proptional to the values in each column.
         Excludes non-numeric data by default.
-
         .. versionadded:: 0.17.1
 
         Parameters
         ----------
         subset: IndexSlice, default None
             a valid slice for ``data`` to limit the style application to
         axis: int
-        color: str
+        color: str (for align='left') or 2-tuple/list (for align='zero', 'mid')
+            If a str is passed, the color is the same for both
+            negative and positive numbers. If 2-tuple/list is used, the
+            first element is the color_negative and the second is the
+            color_positive (eg: ['#d65f5f', '#5fba7d'])
         width: float
             A number between 0 or 100. The largest value will cover ``width``
             percent of the cell's width
+        align : str, default 'left'
+            
+            .. versionadded:: 0.20.0
+            
+            - 'left' : the min value starts at the left of the cell
+            - 'zero' : a value of zero is located at the center of the cell
+            - 'mid' : the center of the cell is at (max-min)/2, or
+                if values are all negative (positive) the zero is aligned
+                at the right (left) of the cell
 
         Returns
         -------
         self : Styler
         """
         subset = _maybe_numeric_slice(self.data, subset)
         subset = _non_reducing_slice(subset)
-        self.apply(self._bar, subset=subset, axis=axis, color=color,
-                   width=width)
+
+        base = 'width: 10em; height: 80%;'
+
+        if not(is_list_like(color)):
+            color = [color, color]
+        elif len(color) == 1:
+            color = [color[0], color[0]]
+        elif len(color) > 2:
+            msg = ("Must pass `color` as string or a list-like"
+                   " of length 2: [`color_negative`, `color_positive`]\n"
+                   "(eg: color=['#d65f5f', '#5fba7d'])")
+            raise ValueError(msg)
+
+        if align == 'left':
+            self.apply(self._bar_left, subset=subset, axis=axis, color=color,
+                       width=width, base=base)
+        elif align == 'zero':
+            self.apply(self._bar_center_zero, subset=subset, axis=axis,
+                       color=color, width=width, base=base)
+        elif align == 'mid':
+            self.apply(self._bar_center_mid, subset=subset, axis=axis,
+                       color=color, width=width, base=base)
```

Comment:
Can you do here an "else" clause to raise a ValueError when a not accepted value for align was passed? and also add a test for it. 
(with playing a bit, I already was bitten by this by trying to pass 'right' and 'center' to align, and thinking that it was not working :-))